File: src/options.c
#include "options.h"

void
options_default(options_t options)
{
  options->tol_relative = 1e-5;
  options->prec_factor = 1.2;
  options->N_beg = 4;
  options->N_end = 16;
  options->N_step = 2;
  options->plot_n = 100;
  options->output = 0;
  options->output_final = 0;
  options->output_time = 0;
}

File: src/julia/Options.jl
mutable struct Options
    tol_relative::Float64
    prec_factor::Float64
    N_beg::Int
    N_end::Int
    N_step::Int
    plot_n::Int
    output::Int32
    output_final::Int32
    output_time::Int32

    function Options()
        opt = new()
        ccall((:options_default, "build/particular_solution"),
              Nothing, (Ref{Options},), opt)
        return opt
    end
end

function Options(opt::Options, Ns::StepRange)
    opt.N_beg = Ns.start
    opt.N_end = Ns.stop
    opt.N_step = Ns.step
    return opt
end

function Options(Ns::StepRange)
    opt = Options()
    options(opt, Ns)
end
